classes/auth/AuthSPSaml.class.php

```diff
@@ -216,6 +216,49 @@ public static function attributes()
         
         return self::$attributes;
     }
+
+
+    public static function ensureLocalIdPMetadata( $entityId, $idp )
+    {
+        $ssp = self::loadSimpleSAML();
+
+
+Logger::error("AAABBB ensureLocalIdPMetadata 0");
+
+        $c = new ReflectionClass("\SimpleSAML\Metadata\MetaDataStorageHandler");
+        $method = $c->getMethod('getMetadataHandler');
+        $mdh = $method->invoke(null);
+Logger::error("AAABBB ensureLocalIdPMetadata 2");
+
+   
+        $sspmd = $mdh->getList('saml20-idp-remote', true );
+Logger::error("AAABBB ensureLocalIdPMetadata 3");
+Logger::error("AAABBB ensureLocalIdPMetadata 3 " . print_r($sspmd,true));
+        $sspmd = $sspmd[$entityId];
+Logger::error("AAABBB ensureLocalIdPMetadata 4 " . print_r($sspmd,true));
+Logger::error("AAABBB ensureLocalIdPMetadata 5 entityId $entityId ");
+Logger::error("AAABBB ensureLocalIdPMetadata 6 " . print_r($sspmd,true));
```

Comment:
Probably the logging needs to be removed before ending up in a release?

Reply:
I would say these logging definitely needs to be removed before ending up in a release.

From the PR description "This is a rough first draft. It needs a combing over and other testing before merge"

These copious debug messages are here because I am debugging the use of this with the help of an admin at an NREN.